mesos/src/cgcloud/mesos/mesos_box.py

```diff
@@ -96,19 +107,36 @@ def _setup_package_repos( self ):
         run('echo "deb http://repos.mesosphere.io/{} {} main"'
             ' | sudo tee /etc/apt/sources.list.d/mesosphere.list'.format( distro, codename ) )
 
+    def _get_iam_ec2_role( self ):
+        role_name, policies = super( MesosBox, self )._get_iam_ec2_role( )
+        role_name += '--' + abreviated_snake_case_class_name( MesosBox )
+        policies.update( dict(
+            ec2_read_only=ec2_read_only_policy,
+            ec2_spark_box=dict( Version="2012-10-17", Statement=[
+                dict( Effect="Allow", Resource="*", Action="ec2:CreateTags" ),
+                dict( Effect="Allow", Resource="*", Action="ec2:CreateVolume" ),
+                dict( Effect="Allow", Resource="*", Action="ec2:AttachVolume" ) ] ) ) )
+        return role_name, policies
+
     def _pre_install_packages( self ):
         super( MesosBox, self )._pre_install_packages( )
         self.__setup_application_user( )
 
+    def _list_packages_to_install( self ):
+        return super( MesosBox, self )._list_packages_to_install( ) + [
+            'mesos=0.21.1-1.1.ubuntu1404', 'docker.io']
```

Comment:
Speaking of which. Maybe we can pull the docker-related stuff into a mixin that John could also use for his Jenkins slave role.